code/controllers/subsystem/ships.dm

```diff
@@ -117,24 +118,24 @@ SUBSYSTEM_DEF(ship)
 		if(S.shield_strength >= initial(S.shield_strength))
 			if(S.shield_strength > starting_shields) broadcast_message("<span class=notice>[faction2prefix(S)] ship ([S.name]) has recharged shields to 100% strength.</span>",notice_sound,S)
 
-	if(!find_broken_ship_components(S))
-		S.next_repair = world.time + S.repair_time
 	if(world.time > S.next_repair && S.repair_time)
 		S.next_repair = world.time + S.repair_time
-		var/datum/ship_component/C
-
-		while(!C && find_broken_ship_components(S)) //pick a broken ship_component to fix
-			C = pick(S.ship_components)
-			if(C.active) C = null
-		if(C)
-			C.active = 1 //fix that shit
-
+		var/list/ship_components = S.ship_components
```

Comment:
Why even store this variable locally?

Oh I know because if it's in a subsystem it *has* to do the most dead horse beating you could possibly imagine.